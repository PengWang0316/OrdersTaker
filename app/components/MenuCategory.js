import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';

import MenuItem from './MenuItem';
import HiddenItemPanel from './HiddenItemPanel';

const flexBox = {
  display: 'flex',
  flexWrap: 'wrap'
};

const styles = {
  root: {
    maxWidth: 1200,
    padding: '10px 20px',
    margin: '15px auto'
  },
  title: {
    fontSize: 23,
    fontWeight: 'bold',
  },
  flexBoxAround: {
    ...flexBox,
    justifyContent: 'space-around'
  },
  flexBoxStart: {
    ...flexBox,
    justifyContent: 'flex-start'
  },
  '@media (max-width: 500px)': {
    flexBoxStart: { justifyContent: 'center' }
  }
};

/**
 * @param {object} items contains item objects.
 * @param {number} itemAmount is the number will be showed at the panel.
 * @return {array} Return a two elements array. The first one is the items will be showed when the second one is the items that will be hidden.
 */
const getSeparatedItems = (items, itemAmount) => {
  const returnArray = [[], []];
  items.forEach((item, index) => {
    if (index < itemAmount) returnArray[0].push(item);
    else returnArray[1].push(item);
  });
  return returnArray;
};

export const MenuCategory = ({ classes, menu, itemAmount }) => {
  const separatedArray = getSeparatedItems(menu.items, itemAmount);
  return (
    <div className={classes.root}>
      <Typography className={classes.title} color="secondary" name={menu.category}>
        {menu.category}
      </Typography>
      <div className={menu.items.length < itemAmount ? classes.flexBoxStart : classes.flexBoxAround}>
        {separatedArray[0].map(item => <MenuItem key={item._id} item={item} />)}
      </div>
      <HiddenItemPanel items={separatedArray[1]} category={menu.category} />
    </div>
  );
};
MenuCategory.propTypes = {
  classes: PropTypes.object.isRequired,
  menu: PropTypes.object.isRequired,
  itemAmount: PropTypes.number.isRequired

};
export default withStyles(styles)(MenuCategory);
