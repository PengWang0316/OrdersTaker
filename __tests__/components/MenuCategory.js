import React from 'react';
import renderer from 'react-test-renderer';

import MenuCategory from '../../app/components/MenuCategory';

jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('../../app/components/MenuItem', () => 'MenuItem');
jest.mock('../../app/components/HiddenItemPanel', () => 'HiddenItemPanel');

describe('MenuCategory', () => {
  test('Snapshot menu.times length > isAmount', () =>
    expect(renderer.create(<MenuCategory {...{ itemAmount: 1, menu: { category: 'category', items: [{ _id: 1 }, { _id: 2 }] }, classes: { root: 'root', title: 'title', flexBoxAround: 'flexBoxAround', flexBoxStart: 'flexBoxStart' } }} />).toJSON())
      .toMatchSnapshot());

  test('Snapshot menu.times length < isAmount', () =>
    expect(renderer.create(<MenuCategory {...{ itemAmount: 5, menu: { category: 'category', items: [{ _id: 1 }, { _id: 2 }] }, classes: { root: 'root', title: 'title', flexBoxAround: 'flexBoxAround', flexBoxStart: 'flexBoxStart' } }} />).toJSON())
      .toMatchSnapshot());
});
