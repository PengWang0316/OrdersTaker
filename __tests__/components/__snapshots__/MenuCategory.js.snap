// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCategory Snapshot menu.times length < isAmount 1`] = `
<div
  className="MenuCategory-root-1 root"
>
  <Typography
    className="MenuCategory-title-2 title"
    color="secondary"
    name="category"
  >
    category
  </Typography>
  <div
    className="MenuCategory-flexBoxStart-4 flexBoxStart"
  >
    <MenuItem
      item={
        Object {
          "_id": 1,
        }
      }
    />
    <MenuItem
      item={
        Object {
          "_id": 2,
        }
      }
    />
  </div>
  <HiddenItemPanel
    category="category"
    items={Array []}
  />
</div>
`;

exports[`MenuCategory Snapshot menu.times length > isAmount 1`] = `
<div
  className="MenuCategory-root-1 root"
>
  <Typography
    className="MenuCategory-title-2 title"
    color="secondary"
    name="category"
  >
    category
  </Typography>
  <div
    className="MenuCategory-flexBoxAround-3 flexBoxAround"
  >
    <MenuItem
      item={
        Object {
          "_id": 1,
        }
      }
    />
  </div>
  <HiddenItemPanel
    category="category"
    items={
      Array [
        Object {
          "_id": 2,
        },
      ]
    }
  />
</div>
`;
